### PR TITLE
Oliver West

### PR DESCRIPTION
**Pull request by WestyCodes**

### CODE REVIEW

File: spec/cohortManager.spec.js [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    // execute
+    const result = cohortManager.createCohort('Cohort 1')
+    // verify
+    expect(result).toEqual(expected)
+  })
+
+  it('Should create a student with a unique ID', () => {
+    // setup
+    const student1 = new Student('John', 'Deere')
+    student1.studentId = 1
```
Comment (dogezen) [inline spec/cohortManager.spec.js]: property for student could be named `id` so you write code like this `student1.id`

File: spec/cohortManager.spec.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    const result = cohortManager.findCohort('Cohort 2')
+    // verify
+    expect(result).toEqual(expected)
+  })
+
+  it('Should search for a cohort by name, but return an error if no cohort found', () => {
+    // setup
+    cohortManager.createCohort('Cohort 1')
+    cohortManager.createCohort('Cohort 2')
+    const expected = `No cohort with this name`
```
Comment (dogezen) [inline spec/cohortManager.spec.js]: refactor to throw new Error and catch these, if you feel like it

File: spec/cohortManager.spec.js [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    const cohort1 = new Cohort('Cohort 1')
+    const student1 = new Student('John', 'Doe')
+    student1.studentId = 1
+    student1.cohort = 'Cohort 1'
+    cohort1.students.push(student1)
+    const expected = cohort1
+    cohortManager.createCohort('Cohort 1')
+    cohortManager.createStudent('John', 'Doe')
+    // execute
+    const result = cohortManager.addStudentToCohort(1, 'Cohort 1')
```
Comment (dogezen) [inline spec/cohortManager.spec.js]: this line would be best written as `cohortManager.addStudentToCohort(student2.id, cohort1.name)` or similar

File: spec/cohortManager.spec.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    // verify
+    expect(result).toEqual(expected)
+  })
+
+  it('Should remove a cohort by name', () => {
+    // setup
+    const cohort1 = new Cohort('Cohort 1')
+    const expected = [cohort1]
+    cohortManager.createCohort('Cohort 1')
```
Comment (dogezen) [inline spec/cohortManager.spec.js:96]: are your cohort names case sensitive? so can I create "Cohort 1" and "cohort 1"? or should I not be allowed to use hte same name?

File: spec/cohortManager.spec.js [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+    cohortManager.createStudent('John', 'Doe')
+    cohortManager.createStudent('Tom', 'Johnson')
+    // execute
+    const result = cohortManager.findStudentByName('John')
+    // verify
+    expect(result).toEqual(expected)
+  })
+})
+
+describe('Cohort', () => {
```
Comment (dogezen) [inline spec/cohortManager.spec.js]: move this into a separate tests file since youre testing cohort

File: spec/cohortManager.spec.js [hunk 6]
```diff
@@ -0,0 +1,218 @@
+const { CohortManager, Cohort, Student } = require('./../src/cohortManager.js')
```
Comment (dogezen) [inline spec/cohortManager.spec.js:1]: youd want some tests for Cohort and Student class, even just to make sure that the constructors save hte right data when an obj is created.

**Review by dogezen — CHANGES_REQUESTED** [review]

File: spec/cohortManager.spec.js (hunk 1 above)
Comment (WestyCodes) [reply spec/cohortManager.spec.js]: done!

File: spec/cohortManager.spec.js (hunk 2 above)
Comment (WestyCodes) [reply spec/cohortManager.spec.js]: done! - all tests/code throw and catch errors.

File: spec/cohortManager.spec.js (hunk 3 above)
Comment (WestyCodes) [reply spec/cohortManager.spec.js]: done!

File: spec/cohortManager.spec.js (hunk 4 above)
Comment (WestyCodes) [reply spec/cohortManager.spec.js:96]: all cohort names are forced into lower case now

File: spec/cohortManager.spec.js (hunk 5 above)
Comment (WestyCodes) [reply spec/cohortManager.spec.js]: done

File: spec/cohortManager.spec.js (hunk 6 above)
Comment (WestyCodes) [reply spec/cohortManager.spec.js:1]: and also done!

**Pull request closed without merging**